packages/evm/src/interpreter.ts

```diff
@@ -259,41 +266,54 @@ export class Interpreter {
   async runStep(): Promise<void> {
     const opInfo = this.lookupOpInfo(this._runState.opCode)
 
-    let gas = BigInt(opInfo.fee)
-    // clone the gas limit; call opcodes can add stipend,
-    // which makes it seem like the gas left increases
-    const gasLimitClone = this.getGasLeft()
-
-    if (opInfo.dynamicGas) {
-      const dynamicGasHandler = (this._evm as any)._dynamicGasHandlers.get(this._runState.opCode)!
-      // This function updates the gas in-place.
-      // It needs the base fee, for correct gas limit calculation for the CALL opcodes
-      gas = await dynamicGasHandler(this._runState, gas, this.common)
-    }
+    let timer: Timer
 
-    if (this._evm.events.listenerCount('step') > 0 || this._evm.DEBUG) {
-      // Only run this stepHook function if there is an event listener (e.g. test runner)
-      // or if the vm is running in debug mode (to display opcode debug logs)
-      await this._runStepHook(gas, gasLimitClone)
+    if (this.profilerOpts?.enabled === true) {
+      timer = this.performanceLogger.startTimer(opInfo.name)
     }
 
-    // Check for invalid opcode
-    if (opInfo.name === 'INVALID') {
-      throw new EvmError(ERROR.INVALID_OPCODE)
-    }
+    let gas = BigInt(opInfo.fee)
 
-    // Reduce opcode's base fee
-    this.useGas(gas, `${opInfo.name} fee`)
-    // Advance program counter
-    this._runState.programCounter++
+    try {
```

Comment:
Why does this require a `try`?  Can't we just check if the timer exists and stop it at the end with out the `try/finally` syntax?

Reply:
It is not `try/finally` because of (internal) timer errors.

The reason is that without try/catch, the inner method can error, for instance, see:

```typescript
      // Check for invalid opcode
      if (opInfo.name === 'INVALID') {
        throw new EvmError(ERROR.INVALID_OPCODE)
      }
```

If this thus throws, the timer is never stopped. (This can also happen within other opcodes, e.g. not enough gas, stack underflow, etc.)

Reply:
Is there intentionally no `catch` block here and if an error is thrown, does that `finally` block still get executed to stop the timer?

Reply:
Yes, `finally` always runs (I had to check this as well)

Reply:
```
try { throw new Error() } finally { console.log(1) }
1
Uncaught Error
```

Reply:
```
try { throw new Error() } catch(e) { throw(e) } finally { console.log(1) }
1
Uncaught Error
```

Reply:
Interesting, learned something new today.

Reply:
So this is very nice, you can thus add code which always runs even if it errors or not (even if catch block re-raises the error). So this is a nice pattern if you setup something which you **always** want to cleanup after. Geth has this natively with `defer`, I wanted this in JS too but apparently this is the way to do that :smile: